features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/parameterUtils.js

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+/**
+ * Defining the supported data types by the OAS spec versions.
+ *
+ * 2.0 : https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#dataTypeFormat
+ * 3.x :https://swagger.io/docs/specification/data-models/data-types/
+ * */
+const supportedDataTypes = {
```

Comment:
IMO , since parameters always come under an operation, (unless referred) it's better to keep one util file (operations.utils). to keep the util methods, or even one util file for the entire resource page, otherwise future deves will get confused where to put what and where to look for what? WDYT ?

Reply:
Yes it will make the utils file bit large, since it contains segregated until methods I think the number of lines in the file would be ok